X2WOTCCommunityPromotionScreen/Src/X2WOTCCommunityPromotionScreen/Classes/CPS_UIArmory_PromotionHero.uc

```diff
@@ -1875,3 +1864,153 @@ simulated function AddChildTweenBetween(string ChildPath, String Prop, float Sta
 
 	MC.EndOp();
 }
+
+// Start Issue #28
+// Replace parent class functions that serve no purpose in CPS, but may get called anyway, generating log errors.
+simulated function UpdateClassRowSelection() {}
+
+simulated function Hide()
+{
+	super(UIScreen).Hide();
+
+	if (NavHelp != none)
+	{
+		NavHelp.Hide();
+	}
+}
+
+simulated function OnReceiveFocus()
+{
+	local XComHQPresentationLayer HQPres;
+
+	super(UIArmory).OnReceiveFocus();
+
+	HQPres = XComHQPresentationLayer(Movie.Pres);
+
+	if(HQPres != none)
+	{
+		if(bAfterActionPromotion) //If the AfterAction screen is running, let it position the camera
+			HQPres.CAMLookAtNamedLocation(AfterActionScreen.GetPromotionBlueprintTag(UnitReference), `HQINTERPTIME);
+		else
+			HQPres.CAMLookAtNamedLocation(CameraTag, `HQINTERPTIME);
+	}
+
+	UpdateNavHelp();
+
+	Columns[m_iCurrentlySelectedColumn].OnReceiveFocus();
+}
+
+private simulated function bool OnUnrealCommand_UIArmory_PromotionHero_Override(int cmd, int arg)
+{
+	local bool bHandled;
+
+	if (!CheckInputIsReleaseOrDirectionRepeat(cmd, arg))
+	{
+		return false;
+	}
+
+	bHandled = true;
+	
+	bHandled = Columns[m_iCurrentlySelectedColumn].OnUnrealCommand(cmd, arg); // bsg-nlong (1.25.17): Send the input to the column first and see if it can consume it
+	if (bHandled) return true;
+
+	switch(cmd)
+	{
+	case class'UIUtilities_Input'.const.FXS_DPAD_RIGHT:
+	case class'UIUtilities_Input'.const.FXS_VIRTUAL_LSTICK_RIGHT:
+		SelectNextColumn();
+		bHandled = true;
+		break;
+	case class'UIUtilities_Input'.const.FXS_DPAD_LEFT :
+	case class'UIUtilities_Input'.const.FXS_VIRTUAL_LSTICK_LEFT :
+		SelectPrevColumn();
+		bHandled = true;
+		break;
+	default:
+		bHandled = false;
+		break;
+	}
+
+	return bHandled || OnUnrealCommand_OnUnrealCommand_UIArmory_Promotion_Override(cmd, arg);
+}
+
+private simulated function bool OnUnrealCommand_OnUnrealCommand_UIArmory_Promotion_Override(int cmd, int arg)
```

Comment:
Do we really need double "OnUnrealCommand" here?

Reply:
Nope, that's a typo, will fix.